cloud/vsphere/clusteractuator.go

```diff
@@ -61,6 +70,91 @@ func (vc *ClusterActuator) Reconcile(cluster *clusterv1.Cluster) error {
 		glog.Infof("Error setting the Load Balancer members for the cluster: %s", err)
 		return err
 	}
+	// Check if the target kuberenetes is ready or not, and update the ProviderStatus if change is detected
+	err = vc.updateK8sAPIStatus(cluster)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (vc *ClusterActuator) updateK8sAPIStatus(cluster *clusterv1.Cluster) error {
+	currentClusterAPIStatus, err := vc.getClusterAPIStatus(cluster)
```

Comment:
Is this call necessary here?

The below call to `updateClusterAPIStatus` will also [call](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/63/files#diff-437bea91b22d029880c4caf3e100903bR135) `getClusterAPIStatus`, and this will incur in a double ssh to get the kubeconfig. 

Also, by returning an error here, it means the status will not be updated when it's created.